### DISCUSSION TITLE
Icons missing after going from 1.26 -> 1.30.5 and cant create new logins/entries

### DISCUSSION THREAD

**Discussion by fontexD** [dani-garcia/vaultwarden#4470] [Q&A]
we upgraded from 1.26 to 1.30.5 but alot of icons is missing all over the place, cant really figure out why

![icons](https://github.com/dani-garcia/vaultwarden/assets/87015443/3364b5df-3fdf-451a-960e-87d48d46f54e)

**Comment by BlackDex** [comment]
You probably have some caching or security active which blocks these items, or doesn't update the new resources.
In the browser use the F12 option to see what happens and what the reason is those items will not load.

**Reply by fontexD** [reply]
funny thing is nothing is reportet wrong in the dev tools console / applications 
and i cant create new items, and if i manage to find thje item and try create a new org and items in there i get i dont have permissions even thoug im owner ? 

tried incogniito mode aswell to make sure its not a cache issue

**Reply by BlackDex** [reply]
Then it might be ad blockers or other extensions which block those items. Be sure to check both the console and network tab.

Also, try to enable/check the `Disable cache` option.

**Reply by fontexD** [reply]
no ad blocker no extensions nothing ima try the cache thing, crazy if that makes issues for both icons and the option to create new stuff getting api error 400 :O

**Reply by fontexD** [reply]
![cache2](https://github.com/dani-garcia/vaultwarden/assets/87015443/abebdc55-ec09-4d7d-83ae-8b282bac0f63)
![cache](https://github.com/dani-garcia/vaultwarden/assets/87015443/788879ca-5f12-4178-b38b-5df8d2995315)
*
![error](https://github.com/dani-garcia/vaultwarden/assets/87015443/2e616d80-dd05-4e3f-ba68-9a4a66e579e6)

**Reply by BlackDex** [reply]
You probably have MFA/2FA enabled, and that causes the 400 message.

The other messages do not seem to be telling my anything useful.
Also, i actually think that version does not have those icon's there for some reason.

The new web-vault does seem to have it again, but that layout is totally different.

**Reply by fontexD** [reply]
i have 2mfa enabled for all accounts yes, which i prefer to have for secruity reasons, alawyass been using this, how do i fix this so i can keep using 2mfa ? worked perfect in 1.26

**Reply by BlackDex** [reply]
Also, I'm using the same version btw and i see this.
![image](https://github.com/dani-garcia/vaultwarden/assets/610450/ed72a043-cfc7-4780-a0c9-747b1d9a0d26)

**Reply by BlackDex** [reply]
> i have 2mfa enabled for all accounts yes, which i prefer to have for secruity reasons, alawyass been using this, how do i fix this so i can keep using 2mfa ? worked perfect in 1.26

The 400 is not an issue, so nothing to worry about, it just works like this.

**Reply by Gerardv514** [reply]
What hardware are you hosting VW on?

**Reply by fontexD** [reply]
Hosting it in k8s using MySQL db been using for 2 years now I think

**Reply by Gerardv514** [reply]
Ok, not the issue I was thinking. There’s a similar issue on synology devices when TLS security is set too high.

**Reply by fontexD** [reply]
also i think its kinda strange that this have been working for 2 years now, going back to early version, but updating to 1.30.5 really cracked something, since we cant post new entires getting this errror
[2024-04-06 15:33:39.285][vaultwarden::api::core::ciphers][ERROR] No rights to modify the collection
[2024-04-06 15:33:39.285][response][INFO] (post_ciphers_admin) POST /api/ciphers/admin => 400 Bad Request

and icons missing , thugt its not so important, right now its more crtiical to et it back working

NOTHING in the hardware / cluster / config of vaulwarden have changed, ONLY the image version .

**Comment by BlackDex** [comment]
Could you try the latest `testing` tagged version? That now has the new web-vault.

**Reply by fontexD** [reply]
Tried with that version, same issue cant add any new entrires/logins in ANY of my boxes, getting no rights to modify this collection, and ALOT of icons missing aswell, but that is kinda not important, but that we cant use it anymore is kinda crticial, never experieced this issue with it before 

this is the log from the app when trying to add a new login to any colection/box (where i am OWNER)

[2024-04-06 15:33:39.285][vaultwarden::api::core::ciphers][ERROR] No rights to modify the collection
[2024-04-06 15:33:39.285][response][INFO] (post_ciphers_admin) POST /api/ciphers/admin => 400 Bad Request

tried going back to 1,26 but the issue with not beeing able to add new logins and icons missing is stll ther, so something when going from 1.26 to 1.30.5 did something but i have no idea

**Reply by BlackDex** [reply]
Is the reverse proxy changed too?
Maybe some security settings were enabled stuff like that.

**Reply by fontexD** [reply]
nothing is changed around the vaultarden i works as the system administratior dev/ops i am responsible for this so i know by 100% nothing have changed around the infrastrucutre that host this

neither the underlaying hardware/servesr the reverseprroxy   / ingress controller ( k8s) storage / config 

nothing

ONLY changed item is image: vaultwarden/server:1.26 -> 1.30.5 

i host around 400 containers in this k8s cluster btw everything else is fine

**Reply by BlackDex** [reply]
What do the logs say then?
There must be something not working correctly.

Also, please post your `/admin/diagnostics` support string.

**Comment by fontexD** [comment]
I restored all data befopre backup / Mysql-DB and the data on ther persistant storage and used image 1.,26 now everything is back to normal

am i missing any CRAZY breaking changes going from this version to 1.30.5 ?  that causes our entire setup to fail with auth/chipers icons ?

**Reply by BlackDex** [reply]
There are no breaking changes. All should just work fine, doesn't matter if you update from 1.10 to 1.30.5 even.
Also what did the logs of 1.30.5 reported during the actions that failed, and during the loading of the vault ui.

There must be something other then Vaultwarden which is causing this, since i would expect a lot more reported issues if this was more common.

**Reply by fontexD** [reply]
### Your environment (Generated via diagnostics page)
* Vaultwarden version: v1.30.5
* Web-vault version: v2024.1.2b
* OS/Arch: linux/x86_64
* Running within a container: true (Base: Debian)
* Environment settings overridden: true
* Uses a reverse proxy: true
* IP Header check: true (X-Real-IP)
* Internet access: true
* Internet access via a proxy: false
* DNS Check: true
* Browser/Server Time Check: true
* Server/NTP Time Check: true
* Domain Configuration Check: true
* HTTPS Check: true
* Database type: MySQL
* Database version: 8.3.0
* Clients used: 
* Reverse proxy and version: 
* Other relevant information: 

### Config (Generated via diagnostics page)
<details><summary>Show Running Config</summary>

**Environment settings which are overridden:** DOMAIN, SIGNUPS_ALLOWED, INVITATIONS_ALLOWED, ADMIN_TOKEN, SMTP_HOST, SMTP_SECURITY, SMTP_PORT, SMTP_FROM, SMTP_FROM_NAME, SMTP_USERNAME, SMTP_PASSWORD


```json
{
  "_duo_akey": null,
  "_enable_duo": false,
  "_enable_email_2fa": false,
  "_enable_smtp": true,
  "_enable_yubico": true,
  "_icon_service_csp": "",
  "_icon_service_url": "",
  "_ip_header_enabled": true,
  "_smtp_img_src": "cid:",
  "admin_ratelimit_max_burst": 3,
  "admin_ratelimit_seconds": 300,
  "admin_session_lifetime": 20,
  "admin_token": "***",
  "allowed_iframe_ancestors": "",
  "attachments_folder": "config/attachments",
  "auth_request_purge_schedule": "30 * * * * *",
  "authenticator_disable_time_drift": false,
  "data_folder": "config",
  "database_conn_init": "",
  "database_max_conns": 10,
  "database_timeout": 30,
  "database_url": "*****://*******************************************************************",
  "db_connection_retries": 15,
  "disable_2fa_remember": false,
  "disable_admin_token": false,
  "disable_icon_download": false,
  "domain": "*****://**********************",
  "domain_origin": "*****://**********************",
  "domain_path": "",
  "domain_set": true,
  "duo_host": null,
  "duo_ikey": null,
  "duo_skey": null,
  "email_attempts_limit": 3,
  "email_change_allowed": true,
  "email_expiration_time": 600,
  "email_token_size": 6,
  "emergency_access_allowed": true,
  "emergency_notification_reminder_schedule": "0 3 * * * *",
  "emergency_request_timeout_schedule": "0 7 * * * *",
  "enable_db_wal": true,
  "event_cleanup_schedule": "0 10 0 * * *",
  "events_days_retain": null,
  "experimental_client_feature_flags": "fido2-vault-credentials",
  "extended_logging": true,
  "helo_name": null,
  "hibp_api_key": null,
  "icon_blacklist_non_global_ips": true,
  "icon_blacklist_regex": null,
  "icon_cache_folder": "config/icon_cache",
  "icon_cache_negttl": 259200,
  "icon_cache_ttl": 2592000,
  "icon_download_timeout": 10,
  "icon_redirect_code": 302,
  "icon_service": "internal",
  "incomplete_2fa_schedule": "30 * * * * *",
  "incomplete_2fa_time_limit": 3,
  "invitation_expiration_hours": 120,
  "invitation_org_name": "Vaultwarden",
  "invitations_allowed": true,
  "ip_header": "X-Real-IP",
  "job_poll_interval_ms": 30000,
  "log_file": null,
  "log_level": "Info",
  "log_timestamp_format": "%Y-%m-%d %H:%M:%S.%3f",
  "login_ratelimit_max_burst": 10,
  "login_ratelimit_seconds": 60,
  "org_attachment_limit": 512000,
  "org_creation_users": "",
  "org_events_enabled": false,
  "org_groups_enabled": false,
  "password_hints_allowed": true,
  "password_iterations": 100000,
  "push_enabled": false,
  "push_identity_uri": "https://identity.bitwarden.com",
  "push_installation_id": "***",
  "push_installation_key": "***",
  "push_relay_uri": "https://push.bitwarden.com",
  "reload_templates": false,
  "require_device_email": false,
  "rsa_key_filename": "config/rsa_key",
  "send_purge_schedule": "0 5 * * * *",
  "sendmail_command": null,
  "sends_allowed": true,
  "sends_folder": "config/sends",
  "show_password_hint": false,
  "signups_allowed": true,
  "signups_domains_whitelist": "",
  "signups_verify": false,
  "signups_verify_resend_limit": 6,
  "signups_verify_resend_time": 3600,
  "smtp_accept_invalid_certs": false,
  "smtp_accept_invalid_hostnames": false,
  "smtp_auth_mechanism": "Login",
  "smtp_debug": false,
  "smtp_embed_images": true,
  "smtp_explicit_tls": null,
  "smtp_from": "****************************",
  "smtp_from_name": "vaultwarden",
  "smtp_host": "***********************",
  "smtp_password": "***",
  "smtp_port": 25,
  "smtp_security": "starttls",
  "smtp_ssl": null,
  "smtp_timeout": 15,
  "smtp_username": "****************************",
  "templates_folder": "config/templates",
  "tmp_folder": "config/tmp",
  "trash_auto_delete_days": null,
  "trash_purge_schedule": "0 5 0 * * *",
  "use_sendmail": false,
  "use_syslog": false,
  "user_attachment_limit": null,
  "user_send_limit": null,
  "web_vault_enabled": true,
  "web_vault_folder": "web-vault/",
  "websocket_address": "0.0.0.0",
  "websocket_enabled": false,
  "websocket_port": 3012,
  "yubico_client_id": null,
  "yubico_secret_key": null,
  "yubico_server": null
}
```
</details>

**Reply by fontexD** [reply]
startup log

/--------------------------------------------------------------------\
|                        Starting Vaultwarden                        |
|                           Version 1.30.5                           |
|--------------------------------------------------------------------|
| This is an *unofficial* Bitwarden implementation, DO NOT use the   |
| official channels to report bugs/features, regardless of client.   |
| Send usage/configuration questions or feature requests to:         |
|   https://github.com/dani-garcia/vaultwarden/discussions or        |
|   https://vaultwarden.discourse.group/                             |
| Report suspected bugs/issues in the software itself at:            |
|   https://github.com/dani-garcia/vaultwarden/issues/new            |
\--------------------------------------------------------------------/

[INFO] Using saved config from `config/config.json` for configuration.

[WARNING] The following environment variables are being overridden by the config.json file.
[WARNING] Please use the admin panel to make changes to them:
[WARNING] DOMAIN, SIGNUPS_ALLOWED, INVITATIONS_ALLOWED, ADMIN_TOKEN, SMTP_HOST, SMTP_SECURITY, SMTP_PORT, SMTP_FROM, SMTP_FROM_NAME, SMTP_USERNAME, SMTP_PASSWORD

[NOTICE] You are using a plain text `ADMIN_TOKEN` which is insecure.
Please generate a secure Argon2 PHC string by using `vaultwarden hash` or `argon2`.
See: https://github.com/dani-garcia/vaultwarden/wiki/Enabling-admin-page#secure-the-admin_token

[2024-04-06 16:07:39.056][start][INFO] Rocket has launched from http://0.0.0.0:80
[2024-04-06 16:12:32.407][request][INFO] GET /api/config
[2024-04-06 16:12:32.408][response][INFO] (config) GET /api/config => 200 OK
[2024-04-06 16:12:37.646][request][INFO] GET /api/devices/knowndevice
[2024-04-06 16:12:37.650][response][INFO] (get_known_device) GET /api/devices/knowndevice => 200 OK
[2024-04-06 16:12:40.711][request][INFO] POST /identity/accounts/prelogin
[2024-04-06 16:12:40.713][response][INFO] (prelogin) POST /identity/accounts/prelogin => 200 OK
[2024-04-06 16:12:40.972][request][INFO] POST /identity/connect/token
[2024-04-06 16:12:41.037][error][ERROR] 2FA token not provided
[2024-04-06 16:12:41.037][response][INFO] (login) POST /identity/connect/token => 400 Bad Request
[2024-04-06 16:12:51.434][request][INFO] POST /identity/connect/token
[2024-04-06 16:12:51.763][vaultwarden::api::identity][INFO] User <user> logged in successfully. IP: 192.168.1.118
[2024-04-06 16:12:51.763][response][INFO] (login) POST /identity/connect/token => 200 OK
[2024-04-06 16:12:54.587][request][INFO] GET /api/config
[2024-04-06 16:12:54.587][response][INFO] (config) GET /api/config => 200 OK
[2024-04-06 16:12:57.082][request][INFO] POST /identity/connect/token
[2024-04-06 16:12:57.091][response][INFO] (login) POST /identity/connect/token => 200 OK
[2024-04-06 16:12:57.115][request][INFO] GET /api/sync?excludeDomains=true
[2024-04-06 16:12:57.381][response][INFO] (sync) GET /api/sync?<data..> => 200 OK
[2024-04-06 16:12:57.616][request][INFO] GET /api/config
[2024-04-06 16:12:57.617][response][INFO] (config) GET /api/config => 200 OK
[2024-04-06 16:12:57.643][request][INFO] GET /notifications/hub?access_token=eyJ0eXAiOigKV1QiL
[2024-04-06 16:12:57.643][vaultwarden::api::notifications][INFO] Accepting Rocket WS connection from 192.168.1.118
[2024-04-06 16:12:57.643][response][INFO] (websockets_hub) GET /notifications/hub?<data..> => 200 OK
[2024-04-06 16:13:09.430][request][INFO] GET /api/ciphers/organization-details?organizationId=ba339c66-52f7-4

**Reply by fontexD** [reply]
there is alot of wierd stuff going oin also when i refresh and it ask me for my pass to re-log

Logget in as  myuyseer@mydomain.dk on __$2__.

like why does it say on "__$2__."


and hen i click to add a new item it can ONLY select my own box

and when i try to add something via the orignisation i get i cant


going back o 1.26 everything is fine all working

**Reply by fontexD** [reply]
i got something now i tried using version 1.29 now i get this error

[2024-04-06 16:20:57.402][panic][ERROR] thread 'rocket-worker-thread' panicked at 'Error loading ciphers: DatabaseError(Unknown, "Illegal mix of collations (utf8mb4_0900_ai_ci,IMPLICIT) and (utf8mb4_unicode_ci,IMPLICIT) for operation '='")': src/db/models/cipher.rs:636

and it dosent seem to load any data into vaultwareden

**Reply by BlackDex** [reply]
See: https://github.com/dani-garcia/vaultwarden/wiki/Using-the-MariaDB-%28MySQL%29-Backend#foreign-key-errors-collation-and-charset
